Okay, I've run into a situation where a major version bump will be required.

Consider the following `Cargo.toml`:

```
[package]
authors = ["Jorge Aparicio <jorge@japaric.io>"]
name = "msp430-quickstart"
version = "0.1.0"

[dependencies.msp430]
version = "0.1.0"

[dependencies.msp430-rt]
version = "0.1.0"

[dependencies.msp430-rtfm]
git = "https://github.com/japaric/msp430-rtfm"

[dependencies.msp430g2553]
features = ["rt"]
version = "0.1.0"

[dependencies.panic-msp430]
version = "0.1.0"

[profile.release]
lto = true

[patch.crates-io]
msp430 = { git = "https://github.com/rust-embedded/msp430", branch = "rt-up" }
```

Compiling with this manifest leads to the following errors:

```
$ xargo build --release --target=msp430-none-elf --example=timer
error[E0308]: mismatched types
  --> examples\timer.rs:15:54
   |
15 |         let wdt = msp430g2553::WATCHDOG_TIMER.borrow(&cs);
   |                                                      ^^^ expected struct `bare_metal::CriticalSection`, found reference
   |
   = note: expected type `&bare_metal::CriticalSection`
              found type `&&msp430::interrupt::CriticalSection`

error[E0308]: mismatched types
  --> examples\timer.rs:21:40
   |
21 |         let port_1_2 = PORT_1_2.borrow(cs);
   |                                        ^^ expected struct `bare_metal::CriticalSection`, found struct `msp430::interrupt::CriticalSection`
   |
   = note: expected type `&bare_metal::CriticalSection`
              found type `&msp430::interrupt::CriticalSection`
note: Perhaps two different versions of crate `bare_metal` are being used?
  --> examples\timer.rs:21:40
   |
21 |         let port_1_2 = PORT_1_2.borrow(cs);
   |                                        ^^

error[E0308]: mismatched types
  --> examples\timer.rs:27:41
   |
27 |         let clock = SYSTEM_CLOCK.borrow(cs);
   |                                         ^^ expected struct `bare_metal::CriticalSection`, found struct `msp430::interrupt::CriticalSection`
   |
   = note: expected type `&bare_metal::CriticalSection`
              found type `&msp430::interrupt::CriticalSection`
note: Perhaps two different versions of crate `bare_metal` are being used?
  --> examples\timer.rs:27:41
   |
27 |         let clock = SYSTEM_CLOCK.borrow(cs);
   |                                         ^^

error[E0308]: mismatched types
  --> examples\timer.rs:31:38
   |
31 |         let timer = TIMER0_A3.borrow(cs);
   |                                      ^^ expected struct `bare_metal::CriticalSection`, found struct `msp430::interrupt::CriticalSection`
   |
   = note: expected type `&bare_metal::CriticalSection`
              found type `&msp430::interrupt::CriticalSection`
note: Perhaps two different versions of crate `bare_metal` are being used?
  --> examples\timer.rs:31:38
   |
31 |         let timer = TIMER0_A3.borrow(cs);
   |                                      ^^

error[E0308]: mismatched types
  --> examples\timer.rs:49:38
   |
49 |         let timer = TIMER0_A3.borrow(cs);
   |                                      ^^ expected struct `bare_metal::CriticalSection`, found struct `msp430::interrupt::CriticalSection`
   |
   = note: expected type `&bare_metal::CriticalSection`
              found type `&msp430::interrupt::CriticalSection`
note: Perhaps two different versions of crate `bare_metal` are being used?
  --> examples\timer.rs:49:38
   |
49 |         let timer = TIMER0_A3.borrow(cs);
   |                                      ^^

error[E0308]: mismatched types
  --> examples\timer.rs:52:40
   |
52 |         let port_1_2 = PORT_1_2.borrow(cs);
   |                                        ^^ expected struct `bare_metal::CriticalSection`, found struct `msp430::interrupt::CriticalSection`
   |
   = note: expected type `&bare_metal::CriticalSection`
              found type `&msp430::interrupt::CriticalSection`
note: Perhaps two different versions of crate `bare_metal` are being used?
  --> examples\timer.rs:52:40
   |
52 |         let port_1_2 = PORT_1_2.borrow(cs);
   |                                        ^^

error: aborting due to 6 previous errors

For more information about this error, try `rustc --explain E0308`.
error: Could not compile `msp430-quickstart`.

To learn more, run the command again with --verbose.
```

In other words, two incompatible copies of `bare_metal` (from `msp430g2553` and `msp430` respectively) are kept around at the same time, which will break any user code using the `borrow` API once `v0.1.1` is published.

I have not updated dependent crates yet so **please do not merge and please do not publish yet!**

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -7,15 +7,15 @@ and this project adheres to [Semantic Versioning](http://semver.org/).
 
 ## [Unreleased]
 
-## [v0.1.1] - 2019-12-22
+## [v0.2.0] - 2019-12-26
 
 Updated bare_metal to 0.2.x. The crate's API is compatible with 0.1.0, but
-using bare_metal 0.2.x consistently decreases multiple versions of the same
-crate in the dependency graph.
+using bare_metal v0.2.x causes incompatible type errors with device crates
+(PACs) using bare_metal v0.1.x. This requires a major version bump to fix.
 
 ## v0.1.0 - 2017-07-22
 
 Initial release.
 
-[Unreleased]: https://github.com/pftbest/msp430/compare/v0.1.1...HEAD
-[v0.1.1]: https://github.com/pftbest/msp430/compare/v0.1.0...v0.1.1
+[Unreleased]: https://github.com/pftbest/msp430/compare/v0.2.0...HEAD
+[v0.2.0]: https://github.com/pftbest/msp430/compare/v0.1.0...v0.2.0

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -7,7 +7,7 @@ keywords = ["msp430", "interrupt", "register", "peripheral"]
 license = "MIT OR Apache-2.0"
 name = "msp430"
 repository = "https://github.com/pftbest/msp430"
-version = "0.1.1"
+version = "0.2.0"
 
 [dependencies.bare-metal]
 features = ["const-fn"]

